core/state/trie_prefetcher.go

```diff
@@ -188,15 +260,16 @@ func (p *triePrefetcher) trie(root common.Hash) Trie {
 	if p.fetches != nil {
 		trie := p.fetches[root]
 		if trie == nil {
-			p.deliveryMissMeter.Mark(1)
 			return nil
 		}
 		return p.db.CopyTrie(trie)
 	}
-	// Otherwise the prefetcher is active, bail if no trie was prefetched for this root
-	fetcher := p.fetchers[root]
+
+	trieChan := make(chan *subfetcher)
+	defer func() { close(trieChan) }()
+	p.trieChan <- &trieMsg{root, trieChan}
```

Comment:
as above, should copy `triechan` if want to use this way for concurrency access

Reply:
it is ok as noted above